### PR TITLE
Use threads more on startup to improve load time

### PR DESCRIPTION
**Pull request by unknownbrackets**
This uses threads a bit better for a few things, which seem to be improvements overall:
 * Initial memory memset.
 * Memory stick initial size calculation.
 * Relocation processing.

Additionally, for GE frame dumps, uses threads to find previous instances of the data being pushed, which improves the time it takes to create a frame dump.

Ultimately this only shaves milliseconds off load in most cases, but the changes are pretty simple and faster is faster.

-[Unknown]

### CODE REVIEW

**Comment by hrydgard** [comment]
I have a big rewrite of PPSSPP's threading primitives sitting idle in a branch, probably should get that in soon so I won't need to port more things :)  But this seems fine. I'll look at it more in detail later.

File: GPU/Debugger/Record.cpp [hunk 1]
```diff
@@ -166,34 +168,44 @@ static const u8 *mymemmem(const u8 *haystack, size_t off, size_t hlen, const u8
...
-		p++;
 		alignp();
-	}
+		while (p <= pend && !result.load()) {
+			p = (const u8 *)memchr(p, first, pend - p + 1);
+			if (!p) {
+				return;
+			}
+			if (poffset() == 0 && !memcmp(p, needle, nlen)) {
+				result.store(p);
```
Comment (hrydgard) [inline GPU/Debugger/Record.cpp:199]: So the first thread that finds a result "wins", and it's no longer deterministic that it's the first one?

All the other changes in thie PR are easy to reason about and I'm happy to merge, but this one feels a little scary, I have to say.

Comment (unknownbrackets) [reply GPU/Debugger/Record.cpp:199]: Hm, it really only needs to be any (it's essentially an unlimited backreference), but probably better to make it deterministic indeed.  Really, the most efficient thing might be for it to always be the *last* one, but I don't want to have to search through all the memory to find it.

-[Unknown]

Comment (hrydgard) [reply GPU/Debugger/Record.cpp:199]: oh right, this is for compression. Still, determinism feels better :)

Comment (unknownbrackets) [reply GPU/Debugger/Record.cpp]: Yep, agreed.  Sorry got sidetracked and forgot about this temporarily.  Updated.

-[Unknown]

**Comment by twinaphex** [comment]
I just want to give a headsup that this causes many libretro targets to fail, see here -

https://git.libretro.com/libretro/ppsspp/-/pipelines/20550

**Pull request merged**